### PR TITLE
feat(egh): add slack notification for post create

### PR DESCRIPTION
**Pull request by zacjones93**
gif

### CODE REVIEW

File: apps/egghead/src/lib/posts-query.ts [hunk 1]
```diff
@@ -211,6 +214,13 @@ export async function createPost(input: NewPost) {
 	})
 
 	if (post) {
+		await inngest.send({
+			name: POST_CREATED_EVENT,
+			data: {
+				post: await post,
+			},
+		})
+
```
Comment (zacjones93) [inline apps/egghead/src/lib/posts-query.ts:223]: The one thing I'm not sure about is awaiting the post here in `post-query.ts`, when I sent over the post as a promise the data was undefined 🤔

File: apps/egghead/src/lib/posts.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ export const PostSchema = ContentResourceSchema.merge(
 			github: z.string().nullish(),
 			gitpod: z.string().nullish(),
 		}),
-		tags: z.array(z.any()),
+		tags: z.array(z.any()).nullish(),
```
Comment (zacjones93) [inline apps/egghead/src/lib/posts.ts:59]: noticed while deleting test posts that there was an issue with the zod schema for posts with no tags

File: apps/egghead/src/inngest/functions/notify-slack-for-post.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import { POST_CREATED_EVENT } from '../events/post-created'
+
+export const notifySlack = inngest.createFunction(
+	{
+		id: `notify-slack`,
+		name: `Notify Slack`,
+	},
+	{
+		event: POST_CREATED_EVENT,
+	},
```
Comment (zacjones93) [inline apps/egghead/src/inngest/functions/notify-slack-for-post.ts:15]: This should be batched in inngest

https://www.inngest.com/docs/guides/batching

File: apps/egghead/src/lib/posts-query.ts (hunk 1 above)
Comment (joelhooks) [reply apps/egghead/src/lib/posts-query.ts:223]: `writeNewPostToDatabase` should probably be awaited above to be more "normal"

File: apps/egghead/src/lib/posts.ts (hunk 2 above)
Comment (joelhooks) [reply apps/egghead/src/lib/posts.ts:59]: better to have a default empty array generally

**Review by joelhooks — APPROVED** [review]

**Pull request merged**